connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java

```diff
@@ -1133,6 +1137,111 @@ public void setTargetState(String connName, TargetState state, Callback<TargetSt
         }
     }
 
+    /**
+     * Get the current offsets for a connector. This method is asynchronous and the passed callback is completed when the
+     * request finishes processing.
+     *
+     * @param connName the name of the connector whose offsets are to be retrieved
+     * @param connectorConfig the connector's configurations
+     * @param cb callback to invoke upon completion of the request
+     */
+    public void connectorOffsets(String connName, Map<String, String> connectorConfig, Callback<ConnectorOffsets> cb) {
+        executor.submit(() -> {
+            String connectorClassOrAlias = connectorConfig.get(ConnectorConfig.CONNECTOR_CLASS_CONFIG);
+            ClassLoader connectorLoader = plugins.connectorLoader(connectorClassOrAlias);
+            Connector connector;
+
+            try (LoaderSwap loaderSwap = plugins.withClassLoader(connectorLoader)) {
+                connector = plugins.newConnector(connectorClassOrAlias);
```

Comment:
Nit: we don't need to predeclare the `connector` variable:

```suggestion

            try (LoaderSwap loaderSwap = plugins.withClassLoader(connectorLoader)) {
                Connector connector = plugins.newConnector(connectorClassOrAlias);
```

Reply:
Whoops, thanks, that was a remnant from before we wrapped this whole section with the thread context classloader switch.